kiwi/xml_state.py

```diff
@@ -958,18 +958,28 @@ def get_build_type_bootloader_name(self) -> str:
         return bootloader.get_name() if bootloader else \
             Defaults.get_default_bootloader()
 
-    def get_build_type_bootloader_console(self) -> Optional[str]:
+    def get_build_type_bootloader_console(self) -> List[str]:
         """
         Return bootloader console setting for selected build type
 
-        :return: console string
+        :return:
+            list of console settings for output (first element)
+            and input (second element)
 
-        :rtype: str
+        :rtype: list
         """
+        result = ['', '']
         bootloader = self.get_build_type_bootloader_section()
         if bootloader:
-            return bootloader.get_console()
-        return None
+            combined_console = bootloader.get_console()
+            if combined_console:
+                console_out, *console_in = combined_console.split(' ')[:2]
```

Comment:
just out of curiosity for splitting in two I like the _partition_ [method](https://docs.python.org/3/library/stdtypes.html#str.partition). In general I used to find it easier to consume specially regarding some corner cases.

Reply:
yeah true ```combined_console.partition(' ')``` would be the smarter coding

Reply:
I remember though I had a reason to do it that way. The XML schema handles the console setting as a collection of allowed strings (console, gfxterm, etc) separated by a space. I did not found an easy way to tell the schema that exactly two tuples are allowed. Thus you can set something like

```
"console gfxterm console"
```

In this case we want to use only the first tuple and forgot about the rest. I also documented it that way.
So the code needs to care for that and also for the fact if only one setting is provided the tuple still has to be complete. I solved this via

```
x="console gfxterm console"
a, *c = x.split(' ')[:2]
c = a if not c else c[0]
```

If I use ```partition``` for this purpose I have to add more code to handle the third portion of partition.
I'm sure there are more ways to do this nicely but given that I could be sure it will be a space separated string because that's enforced by the schema the split approach looks safe and readable to me